src/join.ts

```diff
@@ -0,0 +1,3 @@
+import { normalize } from './normalize';
+
+export const join = (...parts: string[]) => normalize(parts.map(normalize).join('/'))
```

Comment:
This is quite inconsistent with `path.join`.
```
path.join('/d/a', '..', 'c'); // /d/c
join('/d/a', '..', 'c'); // /d/a/../c
```
:disappointed:

Reply:
@wmhilton going to use `path.join` here.

Reply:
no, you just need to fix `normalize` 😆 it doesn't resolve '..'

Reply:
The question is - is there any point of doing it when `path.join` just works fine? :thinking:

Reply:
How about using path.join for now so that we can try these fixes out, and then could replace path.join with a custom implementation later? I'm not convinced that building out a custom join function that is cross platform and handles all the edge cases will be trivial.

Reply:
> no, we just need to fix normalize laughing it doesn't resolve '..'

I can write some simple regular expression, just not sure I see the point of doing this when `path.join` seems to be meet our current requirements.

Reply:
@marbemac @wmhilton done